examples/pytorch/gnn_explainer/NodeExplainerModule.py

```diff
@@ -0,0 +1,203 @@
+
+        if self.mask_bias:
+            mask_bias = nn.Parameter(th.Tensor(num_edges, 1))
+            nn.init.constant_(mask_bias, 0.0)
+        else:
+            mask_bias = None
+
+        return mask, mask_bias
+
+
+    def create_node_feat_mask(self, node_feat_dim, init_strategy="normal"):
+        """
+        Based on the dimensions of node feature in the computational graph, create a learnable mask of features.
+
+        Parameters
+        ----------
+        node_feat_dim: Integer N, dimensions of node feature
+        init_strategy: String, specify the parameter initialization method
+
+        Returns
+        -------
+        mask: Tensor, in shape of N
+        """
+        mask = nn.Parameter(th.Tensor(node_feat_dim))
+
+        if init_strategy == "normal":
+            std = 0.1
+            with th.no_grad():
+                mask.normal_(1.0, std)
+        elif init_strategy == "constant":
+            with th.no_grad():
+                nn.init.constant_(mask, 0.0)
+        return mask
+
+
+    def forward(self, graph, n_feats):
+        """
+        Calculate prediction results after masking input of the given model.
+
+        Parameters
+        ----------
+        graph: DGLGraph, Should be a sub_graph of the target node to be explained.
+        n_idx: Tensor, an integer, index of the node to be explained.
+
+        Returns
+        -------
+        new_logits: Tensor, in shape of N * Num_Classes
+
+        """
+
+        # Step 1: Mask node feature with the inner feature mask
+        new_n_feats = n_feats * self.node_feat_mask.sigmoid()
+        edge_mask = self.edge_mask.sigmoid()
+
+        # Step 2: Add compute logits after mask node features and edges
+        new_logits = self.model(graph, new_n_feats, edge_mask)
+
+        return new_logits
+
+
+    def _loss(self, pred_logits, pred_label):
+        """
+        Compute the losses of this explainer, which include 6 parts in author's codes:
+        1. The prediction loss between predict logits before and after node and edge masking;
+        2. Loss of edge mask itself, which tries to put the mask value to either 0 or 1;
+        3. Loss of node feature mask itself,  which tries to put the mask value to either 0 or 1;
+        4. L2 loss of edge mask weights, but in sum not in mean;
+        5. L2 loss of node feature mask weights, which is NOT used in the author's codes;
+        6. Laplacian loss of the adj matrix.
+
+        In the PyG implementation, there are 5 types of losses:
+        1. The prediction loss between logits before and after node and edge masking;
+        2. Sum loss of edge mask weights;
+        3. Loss of edge mask entropy, which tries to put the mask value to either 0 or 1;
+        4. Sum loss of node feature mask weights;
+        5. Loss of node feature mask entropy, which tries to put the mask value to either 0 or 1;
+
+        In this implementation, will use the PyG's version, but set the aggregation function of weights
```

Comment:
Please remove the ", but ..." and change to a period. I didn't set the aggregation function to be optional.